lib/Transforms/Instrumentation/ShadowMemSeaDsa.cc

```diff
@@ -604,8 +634,10 @@ void ShadowDsaImpl::visitLoadInst(LoadInst &I) {
     return;
 
   m_B->SetInsertPoint(&I);
-  mkShadowLoad(*m_B, c);
+  CallInst *memUse = mkShadowLoad(*m_B, c);
+  m_shadowLoadToLoad.insert({memUse, &I});
```

Comment:
Add a comment where `m_shadowLoadToLoad` and the other map is defined that they are option.
Factor out the code that uses them, i.e., so that the maps are not used explicitly.

In the future, if for some reason this becomes a bottleneck we might switch to a scheme in which we search 
for the corresponding load/store by traversing the basic block or by looking at some metadata.